src/service/ampdoc-impl.js

```diff
@@ -418,20 +418,47 @@ export class AmpDoc {
     return v == null ? null : v;
   }
 
+  /**
+   * Returns map of an ampdoc's meta name value to content value
+   * @return {!Object<string, string>}
+   */
+  getMeta() {
+    const metaMap = map();
+    if (!this.win.document || !this.win.document.head) {
+      return metaMap;
+    }
+
+    const metas = this.win.document.head.querySelectorAll('meta[name]');
+    iterateCursor(metas, meta => {
+      const content = meta.getAttribute('content');
+      const name = meta.getAttribute('name');
+      if (!name || content === null) {
+        return;
+      }
+
+      // Retain only the first meta content value for a given name, consistent
+      // with getMetaByName.
+      if (metaMap[name] === undefined) {
```

Comment:
If `metaMap[name] !== undefined)`, should we emit `user().warn('AmpDoc', 'Only the first <meta> tag with a given name is retained');`?